### PR TITLE
Add support for checking if a syntax is registered

### PR DESCRIPTION
**Pull request by conorhastings**

### CODE REVIEW

**Comment by wooorm** [comment]
1. why is it needed?
2. could you add docs (edit: done!) and tests?
3. Prism uses different terms compared to highlight.js (e.g., register instead of registerLanguage), so we could use `language` or `syntax` or something else here too, what do you think?

**Comment by conorhastings** [comment]
@wooorm added tests

in regards to your first question, I'd like a way to know if syntax has been previously registered or not.

File: test/index.js [hunk 1]
```diff
@@ -156,6 +156,18 @@ test('.register(grammar)', function (t) {
   t.end();
 });
 
+test.only('.getLanguage(language)', function (t) {
```
Comment (wooorm) [inline test/index.js]: drop the `.only`

**Comment by conorhastings** [comment]
@wooorm updated with tests and coverage passing

File: readme.md [hunk 2]
```diff
@@ -135,6 +136,31 @@ Yields:
...
+
+Check if a [syntax][] has been registered and return syntax if it has.
+
+###### Example
+
+```js
+var refractor = require('refractor/core.js');
+
+refractor.register(require('refractor/lang/markdown.js'));
+refractor.syntax("markdown");
```
Comment (wooorm) [inline readme.md]: This line (L149) is superfluous right?  If so, I can remove it.

**Comment by conorhastings** [comment]
@wooorm fixed example

**Comment by wooorm** [comment]
This looks great!

One more Q, while I was in the process of merging this: the result of `syntax` isn’t useful at all to other people I think. How about a `registered` function instead, that returns a boolean: whether the given language is registered?

Does that solve your problem?

**Comment by conorhastings** [comment]
@wooorm yeah would actually prefer that, just was matching lowlight as much as possible, but i'll update to that

**Comment by wooorm** [comment]
Don’t worry, I’ll do it, I was half way through merging this already!

**Comment by conorhastings** [comment]
@wooorm sounds good, thanks!

**Comment by wooorm** [comment]
Released!

**Comment by conorhastings** [comment]
@wooorm awesome! thanks!

**Pull request closed without merging**